compiler/ilgen/VirtualMachineOperandStack.cpp

```diff
@@ -0,0 +1,197 @@
+/*******************************************************************************
+ *
+ * (c) Copyright IBM Corp. 2000, 2016
+ *
+ *  This program and the accompanying materials are made available
+ *  under the terms of the Eclipse Public License v1.0 and
+ *  Apache License v2.0 which accompanies this distribution.
+ *
+ *      The Eclipse Public License is available at
+ *      http://www.eclipse.org/legal/epl-v10.html
+ *
+ *      The Apache License v2.0 is available at
+ *      http://www.opensource.org/licenses/apache2.0.php
+ *
+ * Contributors:
+ *    Multiple authors (IBM Corp.) - initial implementation and documentation
+ *******************************************************************************/
+
+#include "ilgen/VirtualMachineOperandStack.hpp"
+#include "ilgen/VirtualMachineRegister.hpp"
+#include "compile/Compilation.hpp"
+#include "il/SymbolReference.hpp"
+#include "il/symbol/AutomaticSymbol.hpp"
+#include "ilgen/IlBuilder.hpp"
+#include "ilgen/MethodBuilder.hpp"
+#include "ilgen/TypeDictionary.hpp"
+
+namespace OMR
+{
+
+VirtualMachineOperandStack::VirtualMachineOperandStack(TR::MethodBuilder *mb, int32_t sizeHint,
+   TR::IlType *elementType, OMR::VirtualMachineRegister *stackTopRegister)
+   : VirtualMachineState(),
+   _mb(mb),
+   _stackTopRegister(stackTopRegister),
+   _stackMax(sizeHint),
+   _stackTop(-1),
+   _elementType(elementType)
+   {
+   int32_t numBytes = _stackMax * sizeof(TR::IlValue *);
+   _stack = (TR::IlValue **) TR::comp()->trMemory()->allocateHeapMemory(numBytes);
+   memset(_stack, 0, numBytes);
+
+   // store current operand stack pointer base address so we can use it whenever we need to
+   // to recreate the stack as the interpreter would have
+   mb->Store("OperandStack_base", stackTopRegister->Load(mb));
+
+   _pushAmount = (growsUp()) ? +1 : -1;
+   _stackOffset = stackPtrStartingOffset();
+   }
+
+VirtualMachineOperandStack::VirtualMachineOperandStack(OMR::VirtualMachineOperandStack *other)
+   : VirtualMachineState(),
+   _mb(other->_mb),
+   _stackTopRegister(other->_stackTopRegister),
+   _stackMax(other->_stackMax),
+   _stackTop(other->_stackTop),
+   _elementType(other->_elementType),
+   _pushAmount(other->_pushAmount)
+   {
+   int32_t numBytes = _stackMax * sizeof(TR::IlValue *);
+   _stack = (TR::IlValue **) TR::comp()->trMemory()->allocateHeapMemory(numBytes);
+   memcpy(_stack, other->_stack, numBytes);
+   }
+
+
+// commits the simulated operand stack of values to the virtual machine state
+// the given builder object is the path to the interpreter
```

Comment:
i'll remove that as it's been described in other places as the "typical" use of the Commit() function, and I agree this version sounds a bit confusing.